lib/pagination/adapter.js

```diff
@@ -0,0 +1,50 @@
+const { invalidPageNumber, invalidLimitNumber } = require('./errors');
+
+module.exports = class Adapter {
+  constructor(content, page, limit, req) {
+    this.content = content;
+    this.page = page;
+    this.limit = limit;
+    this.request = req;
+    if (this.page <= 0) throw invalidPageNumber('Invalid Page Number');
```

Comment:
You must check if page and limit are integers too

Reply:
In the future we could define a whole module to process all kind of bad format errors, and things like that, for now, I agree we could have that validation here.